tests/modules/images/webp-uploads/webp-uploads-test.php

```diff
@@ -809,6 +811,166 @@ function( $transforms ) {
 		}
 	}
 
+	/**
+	 * Backup the sources structure alongside the full size
+	 *
+	 * @test
+	 */
+	public function it_should_backup_the_sources_structure_alongside_the_full_size() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+		$this->assertEmpty( get_post_meta( $attachment_id, '_wp_attachment_backup_sizes', true ) );
+		$this->assertEmpty( get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true ) );
+
+		$editor = new WP_Image_Edit( $attachment_id );
+		$editor->rotate_right()->save();
+
+		// Having a thumbnail ensures the process finished correctly.
+		$this->assertTrue( $editor->success() );
+
+		$backup_sizes = get_post_meta( $attachment_id, '_wp_attachment_backup_sizes', true );
+
+		$this->assertNotEmpty( $backup_sizes );
+		$this->assertIsArray( $backup_sizes );
+
+		$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+		$this->assertIsArray( $backup_sources );
+		$this->assertArrayHasKey( 'full-orig', $backup_sources );
+		$this->assertSame( $metadata['sources'], $backup_sources['full-orig'] );
+
+		foreach ( $backup_sizes as $size => $properties ) {
+			$size_name = str_replace( '-orig', '', $size );
+
+			if ( 'full-orig' === $size ) {
+				continue;
+			}
+
+			$this->assertArrayHasKey( 'sources', $properties );
+			$this->assertSame( $metadata['sizes'][ $size_name ]['sources'], $properties['sources'] );
+		}
+
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+	}
+
+	/**
+	 * Restore the sources array from the backup when an image is edited
+	 *
+	 * @test
+	 */
+	public function it_should_restore_the_sources_array_from_the_backup_when_an_image_is_edited() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$metadata      = wp_get_attachment_metadata( $attachment_id );
+		$editor        = new WP_Image_Edit( $attachment_id );
+		$editor->rotate_right()->save();
+		$this->assertTrue( $editor->success() );
+		$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+		$this->assertArrayHasKey( 'full-orig', $backup_sources );
+		$this->assertIsArray( $backup_sources['full-orig'] );
+		$this->assertSame( $metadata['sources'], $backup_sources['full-orig'] );
```

Comment:
Let's add some whitespace here to make it easier to read and follow:
```suggestion
		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
		$metadata      = wp_get_attachment_metadata( $attachment_id );

		$editor = new WP_Image_Edit( $attachment_id );
		$editor->rotate_right()->save();
		$this->assertTrue( $editor->success() );

		$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
		$this->assertArrayHasKey( 'full-orig', $backup_sources );
		$this->assertIsArray( $backup_sources['full-orig'] );
		$this->assertSame( $metadata['sources'], $backup_sources['full-orig'] );
```